IBAnimatable/AnimationChainable.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Created by jason akakpo on 01/01/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+typealias AnimationTuple = (type: AnimationType, configuration: AnimationConfiguration)
+
+struct AnimationConfiguration {
+  let damping: CGFloat
+  let velocity: CGFloat
+  let duration: TimeInterval
+  let delay: TimeInterval
+  let force: CGFloat
+
+}
+
+
+public class AnimationPromise<T: UIView> where T: Animatable {
+
+  private var view: T
+  private var animationList = [AnimationTuple]()
+  private var delayForNextAnimation = 0.0
+  private var completion: AnimatableCompletion?
+
+  init(view: T) {
+    self.view = view
+  }
+
+
```

Comment:
Is your SwitLint up to date? Pretty two following newlines that will fire a warning